### ISSUE TITLE
JavadocPackage: false positive "Missing package-info.java file" on module-info.java

### ISSUE THREAD

**Issue by codeAnqiang-ma** [checkstyle/checkstyle#21199]
Parent Issue #21113

I have read check documentation: https://checkstyle.org/checks/javadoc/javadocpackage.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Note: unlike the TreeWalker cases tracked in the parent issue, `JavadocPackage` is a Checker-level `FileSetCheck` that never parses the file, so this false positive is reproducible on released 13.10.0 — it does not need the #8240 grammar.

```bash
/var/tmp $ find . -not -path "./out*" | sort
.
./config.xml
./src
./src/com
./src/com/example
./src/com/example/api
./src/com/example/api/Api.java
./src/com/example/api/package-info.java
./src/module-info.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="JavadocPackage"/>
</module>

/var/tmp $ cat src/module-info.java
module com.example.app {
    exports com.example.api;
}

/var/tmp $ cat src/com/example/api/package-info.java
/** Public API of the application. */
package com.example.api;

/var/tmp $ cat src/com/example/api/Api.java
package com.example.api;

/** Sample API type. */
public class Api {
}

/var/tmp $ javac -d out src/module-info.java src/com/example/api/package-info.java src/com/example/api/Api.java
# successful compilation, valid JPMS module

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.10.0-all.jar -c config.xml src/
Starting audit...
[ERROR] /var/tmp/src/module-info.java:1: Missing package-info.java file. [JavadocPackage]
Audit done.
Checkstyle ends with 1 errors.
```

The properly documented package `com.example.api` passes. The only violation is on `module-info.java` itself: the check demands a `package-info.java` next to it in the source root.

---

**Describe what you expect in detail.**

No violation. Expected output on the same input:

```
Starting audit...
Audit done.
```

The check documentation says "Checks that each Java **package** has a Javadoc file used for commenting". A modular compilation unit is not a member of any package: per [JLS 7.3](https://docs.oracle.com/javase/specs/jls/se25/html/jls-7.html#jls-7.3), it consists of import declarations and a module declaration and cannot contain a package declaration. So the presence of `module-info.java` in a directory is not evidence of a package that needs a `package-info.java`. The module's own documentation belongs in the Javadoc of the module declaration itself (a dedicated check for that is proposed in #21165).

Root cause: [`JavadocPackageCheck#processFiltered`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java#L90-L115) treats every processed `.java` file as marking a package directory and then requires `package-info.java` in it. Files named `module-info.java` should be skipped, the same way `FileContents#inPackageInfo` special-cases files by the name `package-info.java`.

Currently the stock `sun_checks.xml` (which enables `JavadocPackage`) masks this via its `BeforeExecutionExclusionFileFilter` on `module\-info\.java$` — the workaround that exists only because module declarations could not be parsed (#8240). Once modular compilation units are supported and that exclusion is no longer needed, every modular project running the sun style hits this violation on its mandatory `module-info.java`. Users who enable `JavadocPackage` without that filter already hit it today, as shown above.

I have a minimal fix with a regression test prepared and will open a PR once this issue is approved.

*Disclosure: this report was prepared with AI assistance during a code-correctness review; I reproduced the behavior locally on 13.10.0 and reviewed every claim.*

**Comment by romani** [comment]
@codeAnqiang-ma , please try to use https://checkstyle.org/releasenotes.html#Release_14.0.0

we did update for module-info.java support.

@vivek-0509 , is this false positive that we missed ?

**Comment by romani** [comment]
@codeAnqiang-ma , please send PR.

**Comment by vivek-0509** [comment]
> @vivek-0509 , is this false positive that we missed ?

Yes its a false positive.. we missed it
